packages/firestore/test/unit/specs/spec_builder.ts

```diff
@@ -369,6 +369,17 @@ export class SpecBuilder {
     return this;
   }
 
+  userListensToNamedQuery(
+    name: string,
+    query: Query,
+    readFrom: TestSnapshotVersion
+  ): this {
+    // Note we do not call this.nextStep() here because we reuse userListens().
```

Comment:
I am not sure this needs to be spec tested, and if anything it seems like it should be two steps (one that loads the query, one that listens to it). To me, restoring a Query by name is separate from query execution. It doesn't need to interact with the rest of the system (SyncEngine, RemoteStore). 


The test could be simple unit tests for the Query class:

```
const query1 = query.where('foo', > 1);
storeInBundle('fooQuery',  'querye.where('foo', > 1)}');
const query2 = namedQuery('fooQuery');
expect(query1).to.equal(query2);
```